ch5-catdir/directory.cpp

```diff
@@ -14,7 +14,8 @@ Directory::Directory(const std::string_view& path)
 {
     if (!m_dir)
     {
-        throw std::system_error(errno, std::generic_category(),
+        auto savedErrno = errno;
+        throw std::system_error(savedErrno, std::generic_category(),
```

Comment:
`std::generic_category` c-tor or `operator+` of `filesystem::path` may change `errno` value? Interesting.

Reply:
`operator+` may allocate memory using `malloc`, which may change `errno` value (see _malloc(3)_).